homeassistant/components/fritz/common.py

```diff
@@ -285,3 +306,111 @@ def device_info(self):
             "model": self._fritzbox_tools.model,
             "sw_version": self._fritzbox_tools.sw_version,
         }
+
+
+class FritzBoxBaseSwitch:
+    """Fritz switch base class."""
+
+    def __init__(self, fritzbox_tools: FritzBoxTools, switch_info: SwitchInfo) -> None:
+        """Init Fritzbox port switch."""
+        self.fritzbox_tools: FritzBoxTools = fritzbox_tools
+
+        self._description = switch_info["description"]
+        self._friendly_name = switch_info["friendly_name"]
```

Comment:
I would remove friendly_name from switch_info. The advantage: Can be removed from init of individual switches.

```suggestion
        self._friendly_name = fritzbox_tools.model
```

Reply:
I'm strongly against it, you can have 2 router of the same model in your environment

Reply:
BTW, we use the same naming convention in the rest of the integration ;-)

Reply:
Looks a bit unclean to me. I think it should be handled internally in the classes and not outside.

Reply:
Maybe action `GetInfo` from `UserInterface` service has the needed info.